src/v2i-hub/SpatPlugin/src/SpatPlugin.cpp

```diff
@@ -1,177 +1,148 @@
-	AddMessageFilter<PedestrianMessage>(this, &SpatPlugin::HandlePedestrianDetection);
-	SubscribeToMessages();
-}
-
-SpatPlugin::~SpatPlugin() {
-
-}
-
-void SpatPlugin::UpdateConfigSettings() {
-
-	GetConfigValue<string>("SignalGroupMapping", signalGroupMappingJson, &data_lock);
-	GetConfigValue<string>("Local_IP", localIp, &data_lock);
-	GetConfigValue<string>("Local_UDP_Port", localUdpPort, &data_lock);
-	GetConfigValue<string>("TSC_IP", tscIp, &data_lock);
-	GetConfigValue<string>("TSC_Remote_SNMP_Port", tscRemoteSnmpPort,
-			&data_lock);
-	GetConfigValue<string>("Intersection_Name", intersectionName,
-			&data_lock);
-	GetConfigValue<int>("Intersection_Id", intersectionId, &data_lock);
-
-	isConfigurationLoaded = true;
-}
-
-void SpatPlugin::OnConfigChanged(const char *key, const char *value) {
-	PluginClient::OnConfigChanged(key, value);
-	UpdateConfigSettings();
-}
-
-void SpatPlugin::OnStateChange(IvpPluginState state) {
-	PluginClientClockAware::OnStateChange(state);
-
-	if (state == IvpPluginState_registered) {
-		UpdateConfigSettings();
+	SpatPlugin::SpatPlugin(const std::string &name) :PluginClientClockAware(name) {
+		spatReceiverThread = std::make_unique<tmx::utils::ThreadTimer>(std::chrono::milliseconds(5));
 	}
-}
 
-void SpatPlugin::HandlePedestrianDetection(PedestrianMessage &pedMsg, routeable_message &routeableMsg) {
-	lock_guard<mutex> lock(data_lock);
-	_pedMessage = pedMsg;
-}
-
-int SpatPlugin::Main() {
-
-	int iCounter = 0;
-
-	PLOG(logINFO) << "Waiting for clock initialization";
-
-	// wait for the clock to be initialized and record the time when it is ready
-	getClock()->wait_for_initialization();
-	auto nextSpatTime = getClock()->nowInMilliseconds();
-	PLOG(logINFO) << "Initial nextSpatTime=" << nextSpatTime;
-
-	try {
-		while (_plugin->state != IvpPluginState_error) {
-			// wait to send next message
-			if (isConfigurationLoaded) {
-				if (!isConfigured) {
-
-					usleep(200000);
+	SpatPlugin::~SpatPlugin() {
 
-					int action = sc.getActionNumber();
-
-					/*pthread_mutex_lock(&gSettingsMutex);
-					 std::cout <<  "Get PTLM file specified by configuration settings" << std::endl;
-					 std::string ptlmFile = GetPtlmFile(action);
-					 pthread_mutex_unlock(&gSettingsMutex);
-					 */
-					//if (!ptlmFile.empty())
-					//{
-					_actionNumber = action;
-
-					// sc.spat_message_mutex does not need locked because the thread is not running yet.
-
-					{
-						std::lock_guard<std::mutex> lock(data_lock);
-						string ptlm = "";
-						sc.setConfigs(localIp, localUdpPort, tscIp,
-								tscRemoteSnmpPort, ptlm, intersectionName,
-								intersectionId);
-					}
-					// Start the signal controller thread.
-					sc.Start(signalGroupMappingJson);
-					// Give the spatdata pointer to the message class
-					//smr41.setSpatData(sc.getSpatData());
+	}
 
-					isConfigured = true;
-					//}
+	void SpatPlugin::UpdateConfigSettings() {
+
+		if (this->IsPluginState(IvpPluginState_registered)) {
```

Comment:
Variable naming is inconsistent. Some variables are snake_case while some are camelCase.

Reply:
This is an enumeration and also comes from our legacy code. I am attempting to change all naming in V2X- Hub to Camel Case inline with our C++ guides. How ever this change can not happen all at once. For now it is restricted to newly created variables and classes and variables and classes that are heavily involved with a change set.